test/utils/session-core.test.js

```diff
@@ -0,0 +1,249 @@
+import {
+  getSolidDataset,
+  createContainerAt,
+  mockSolidDatasetFrom,
+  mockThingFrom,
+  createSolidDataset,
+  setThing,
+  saveSolidDatasetInContainer
+} from '@inrupt/solid-client';
+import { expect, vi, it, describe, afterEach, beforeEach } from 'vitest';
+import { getContainerUrl } from '../../src/utils/network/session-helper';
+import { INTERACTION_TYPES } from '../../src/constants';
+
+vi.mock('@inrupt/solid-client', async () => {
+  const actual = await vi.importActual('@inrupt/solid-client');
+  return {
+    ...actual,
+    getSolidDataset: vi.fn((url) => Promise.resolve(mockSolidDatasetFrom(url))),
+    createContainerAt: vi.fn((url) => Promise.resolve(mockSolidDatasetFrom(url))),
+    saveSolidDatasetInContainer: vi.fn((url) => Promise.resolve(mockSolidDatasetFrom(url)))
+  };
+});
+
+const getSolidDatasetRejected = vi.fn(() => Promise.reject(Error('dataset not found')));
+
+const mockPodUrl = 'https://pod.example.com/';
+const mockSolidIdentityProvider = 'https://example.com/';
+let session = {};
+
+describe('setDocAclPermission', () => {
+  beforeEach(() => {
+    session = {
+      fetch: vi.fn(),
+      info: {
+        webId: `${mockPodUrl}profile/card#me`
+      }
+    };
+  });
+  afterEach(() => {
+    vi.clearAllMocks();
+  });
+
+  it("prints out correct documentUrl for Driver's License and webId for other user", () => {
+    const fileType = "Driver's License";
+    const otherPodUsername = 'pod2';
+    const documentUrl = getContainerUrl(session, fileType, INTERACTION_TYPES.SELF);
+    const webId = `https://${otherPodUsername}.${
+      mockSolidIdentityProvider.split('/')[2]
+    }/profile/card#me`;
+
+    expect(documentUrl).toBe('https://pod.example.com/Drivers%20License/');
+    expect(webId).toBe('https://pod2.example.com/profile/card#me');
+  });
+});
+
+const setDocAclForUser = vi.fn(() => Promise.resolve());
+
+describe('setDocContainerAclPermission', () => {
+  beforeEach(() => {
+    session = {
+      fetch: vi.fn(),
+      info: {
+        webId: `${mockPodUrl}profile/card#me`
+      }
+    };
+  });
+  afterEach(() => {
+    vi.clearAllMocks();
+  });
+
+  it('calls setDocAclForUser 4 times', async () => {
+    const otherPodUsername = 'pod2';
+    const permissions = { read: true, write: true, append: true, control: false };
+    const containerUrl = getContainerUrl(session, 'Documents', INTERACTION_TYPES.SELF);
+    const urlsToSet = [
+      containerUrl,
+      `${containerUrl}Bank%20Statement/`,
+      `${containerUrl}Passport/`,
+      `${containerUrl}Drivers%20License/`
+    ];
+
+    const webId = `https://${otherPodUsername}.${
+      mockSolidIdentityProvider.split('/')[2]
+    }/profile/card#me`;
+
+    urlsToSet.forEach(async (url) => {
+      await setDocAclForUser(session, url, 'update', webId, permissions);
```

Comment:
This doesn't call the `setDocAclForUser` from `session-core`. It calls the variable you defined on line 56.

Reply:
The setDocAclForUser here is a mock function

Reply:
Correct, but it probably shouldn't be. You generally only want to mock external services like databases, networks, etc. In our case, we generally should only be mocking the Inrupt services that interact with the solid servers. Our own code generally shouldn't be mocked.